### PR TITLE
Added jekyll assets, updated references and added service worker

### LINKED ISSUE #28: Enable caching for static assets
We should add a service worker to do caching of static resources like js, css, images etc. which will result in faster page load for returning users.
Let me know if this is a good idea, I will be able to help on this.

### PR DESCRIPTION
**Pull request by bhansa**
Hi, 

I have added a service worker to root directory of this project, I have followed this [blog post](https://alxgbsn.co.uk/2017/12/08/adding-a-service-worker-to-jekyll/) for steps to add a service worker to a jekyll site using [jekyll-assets](https://github.com/envygeeks/jekyll-assets). For our assets and code to work with jekyll assets, I have made below few changes:
  - Updated dependencies, added jekyll-assets
  - Update *assets* folder to *_assets* to create an asset pipeline.
  - Update references in our html files as well as data yaml files, used `{% asset asset_name ...attributes %}` syntax instead, more details can be found in the jekyll assets github page.

**Why jekyll asset**?
Since we want to update the cache version, as well as version the of resources/assets like css, js and images. Refer serviceWorker.js file for details.

Let me know if these changes are good, provide suggestions as well.

Thanks,
Bharat

Fixes #28

### CODE REVIEW

**Comment by ananyo2012** [comment]
@bhansa Thanks for your PR. This will add a lot of value to the website. I can't see the Netlify preview. Probably some issue with the asset links. Can you check what's the issue ?

File: serviceWorker.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+                if (response) {
+                    console.log('Serving cached: ', event.request.url);
+                    return response;
+                }
+                console.log('Fetching: ', event.request.url);
+                return fetch(request)
+            })
+    );
+});
+{% endraw %}
```
Comment (palnabarun) [inline serviceWorker.js:83]: nit: new line, please.

File: serviceWorker.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        return;
+    }
+
+    // // For HTML requests, try the network first else fall back to the offline page.
+    // if (request.headers.get('Accept').indexOf('text/html') !== -1) {
+    //     event.respondWith(
+    //         fetch(request).catch(() => caches.match('/offline/'))
+    //     );
+    //     return;
+    // }
```
Comment (palnabarun) [inline serviceWorker.js:68]: nit: please remove the commented if not required.

File: Gemfile [hunk 3]
```diff
@@ -6,3 +6,6 @@ git_source(:github) {|repo_name| "https://github.com/#{repo_name}" }
 
 
 gem "jekyll"
+gem "jekyll-assets"
+gem 'uglifier'
+gem 'sprockets', '~> 3.7'
```
Comment (palnabarun) [inline Gemfile:11]: nit: new line, please.

**Review by palnabarun — CHANGES_REQUESTED** [review]

**Comment by palnabarun** [comment]
Changed base branch to `main` from `master`.

**Comment by SaptakS** [comment]
@bhansa can you take a look at the merge conflicts?

**Comment by bhansa** [comment]
Hi, I tried couple of times to resolve the issue with netlify build and jekyll assets and did not get any success with any of them and I could not spend much time to fix this PR.
I apologize for long delay and I think this PR will not add much value at this point of time, and I should focus on other open issues.
Hence, I am closing this PR.

**Pull request closed without merging**